diff.py

```diff
@@ -433,38 +434,55 @@ def search_map_file(fn_name: str) -> Tuple[Optional[str], Optional[int]]:
 
     try:
         with open(mapfile) as f:
-            lines = f.read().split("\n")
+            lines = f.read()
     except Exception:
         fail(f"Failed to open map file {mapfile} for reading.")
 
-    try:
-        cur_objfile = None
-        ram_to_rom = None
-        cands = []
-        last_line = ""
-        for line in lines:
-            if line.startswith(" .text"):
-                cur_objfile = line.split()[3]
-            if "load address" in line:
-                tokens = last_line.split() + line.split()
-                ram = int(tokens[1], 0)
-                rom = int(tokens[5], 0)
-                ram_to_rom = rom - ram
-            if line.endswith(" " + fn_name):
-                ram = int(line.split()[0], 0)
-                if cur_objfile is not None and ram_to_rom is not None:
-                    cands.append((cur_objfile, ram + ram_to_rom))
-            last_line = line
-    except Exception as e:
-        import traceback
-
-        traceback.print_exc()
-        fail(f"Internal error while parsing map file")
-
-    if len(cands) > 1:
-        fail(f"Found multiple occurrences of function {fn_name} in map file.")
-    if len(cands) == 1:
-        return cands[0]
+    if map_format == 'gnu':
+        lines = lines.split("\n")
+
+        try:
+            cur_objfile = None
+            ram_to_rom = None
+            cands = []
+            last_line = ""
+            for line in lines:
+                if line.startswith(" .text"):
+                    cur_objfile = line.split()[3]
+                if "load address" in line:
+                    tokens = last_line.split() + line.split()
+                    ram = int(tokens[1], 0)
+                    rom = int(tokens[5], 0)
+                    ram_to_rom = rom - ram
+                if line.endswith(" " + fn_name):
+                    ram = int(line.split()[0], 0)
+                    if cur_objfile is not None and ram_to_rom is not None:
+                        cands.append((cur_objfile, ram + ram_to_rom))
+                last_line = line
+        except Exception as e:
+            import traceback
+
+            traceback.print_exc()
+            fail(f"Internal error while parsing map file")
+
+        if len(cands) > 1:
+            fail(f"Found multiple occurrences of function {fn_name} in map file.")
+        if len(cands) == 1:
+            return cands[0]
+    elif map_format == 'mw':
+                                            #    ram   elf rom                                                       object name
```

Comment:
```suggestion
        #                                        ram   elf rom                                                       object name
```

(also, looks a bit misaligned)